pkg/manifests/manifests.go

```diff
@@ -74,36 +74,56 @@ func Get(ns string, sharedCPUs string) (*Manifests, error) {
 		}
 	}
 
-	if ns != "" {
+	// by default, set the namespace to default namespace.
+	// the default namespace can be changed via Optional Parameter Function
+	WithNamespace("")(&mf)
+
+	for _, opt := range opts {
+		opt(&mf)
+	}
+
+	mf.SCC.Users = []string{
+		fmt.Sprintf("system:serviceaccount:%s:%s", mf.SA.Namespace, mf.SA.Name),
+	}
+	if err := mf.SetSharedCPUs(sharedCPUs); err != nil {
+		return nil, err
+	}
+	return &mf, nil
+}
+
+// WithNewNamespace creates new namespace and updates the objects
+func WithNewNamespace(ns string) func(mf *Manifests) {
```

Comment:
this seems like mixing layers (setting properties AND creating objects) and, albeit likely handy, I'd avod it

Reply:
How do you suggest handle the namespace creation?